### PR TITLE
Let storybook be aware of postcss for `.module.css` files

### PR DESCRIPTION
**Pull request by maraisr**
Storybook's `addont-styling` will autowire webpack for us to leverage css modules within our stories, and component files. Pathing the way to start using those as opposed to styled-components. We also prefer this addon now over `storybook-css-modules`.

Huge props to @siddharthkp for starting this work over in https://github.com/primer/react/pull/3405

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Review by langermank — APPROVED** [review]
Approving but would love for @siddharthkp to also approve before merging :)

File: postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/** @type {import('postcss-load-config').Config} */
+module.exports = {
+  plugins: [
+    require('postcss-preset-env')({
+      stage: 2, // https://preset-env.cssdb.org/features/#stage-2
+      features: {
+        'nesting-rules': {noIsPseudoSelector: true},
+        'focus-visible-pseudo-class': false,
+        'logical-properties-and-values': false,
```
Comment (broccolinisoup) [inline postcss.config.js:9]: Curious to hear why they are false 👀

**Review by broccolinisoup — COMMENTED** [review]
This is very exciting! Thanks for this PR. It looks good to me but I am only watching the changes for now as I don't have much context and experience on this area as much as Sid or Katie has :)

File: postcss.config.js (hunk 1 above)
Comment (maraisr) [reply postcss.config.js:9]: Took inspiration from https://github.com/primer/view_components/blob/main/postcss.config.js possibly @langermank can provide more insights

Comment (maraisr) [reply postcss.config.js:9]: I see @simurai/@jonrohan worked on this file and found some comments here: https://github.com/primer/view_components/pull/1638/files#r1032151244

Interesting stuff.

Comment (jonrohan) [reply postcss.config.js:9]: In some of our cases, the output was putting more CSS than we needed for the browsers we supported. I think the focus-visible-pseudo-class was also adding a `.focus` class which was meant to be paired with JS. So it was on, until disabled.

**Review by siddharthkp — APPROVED** [review]
Looks good! Ship it

**Pull request merged**